### ISSUE TITLE
Bug when when model properties file exists

### ISSUE THREAD

**Issue by sfc-gh-pkommini** [Montreal-Analytics/dbt-gloss#98]
**Describe the bug**
I have a dbt `models` directory structure like so:

Model files:
- `models/dir1/sub_dir1/model_a.sql`,
- `models/dir1/sub_dir1/model_b.sql`,
- `models/dir1/sub_dir1/model_c.sql`

Model properties file: `models/dir1/sub_dir1/models_schema.yml` looks like this:
```yml
version: 2
models:
  - name: model_a
    description: "Something something a"

  - name: model_b
    description: "Something something b"

  - name: model_c
    description: "Something something c"
```
When I run this command
```bash
pre-commit run check-model-has-meta-keys --files models/dir1/sub_dir1/model_a.sql
```

And I get this error:
```
Check the model has keys in the meta part................................Failed
- hook id: check-model-has-meta-keys
- exit code: 1

Traceback (most recent call last):
  File "/Users/xxx/.cache/pre-commit/repo4nh1q0_n/py_env-python3.10/bin/check-model-has-meta-keys", line 8, in <module>
    sys.exit(main())
  File "/Users/xxx/.cache/pre-commit/repo4nh1q0_n/py_env-python3.10/lib/python3.10/site-packages/pre_commit_dbt/check_model_has_meta_keys.py", line 99, in main
    return has_meta_key(
  File "/Users/xxx/.cache/pre-commit/repo4nh1q0_n/py_env-python3.10/lib/python3.10/site-packages/pre_commit_dbt/check_model_has_meta_keys.py", line 53, in has_meta_key
    in_schemas = {
  File "/Users/xxx/.cache/pre-commit/repo4nh1q0_n/py_env-python3.10/lib/python3.10/site-packages/pre_commit_dbt/check_model_has_meta_keys.py", line 53, in <setcomp>
    in_schemas = {
  File "/Users/xxx/.cache/pre-commit/repo4nh1q0_n/py_env-python3.10/lib/python3.10/site-packages/pre_commit_dbt/utils.py", line 199, in get_model_schemas
    with open(yml_file, "r") as file:
FileNotFoundError: [Errno 2] No such file or directory: 'project-name/models/dir1/sub_dir1/models_schema.yml'
```

But I do have that file with the structure as seen above. What am I doing wrong?

If I either remove or comment out the models a, b, c from the properties file, then it works and passes the `check-model-has-meta-keys` check.

**To Reproduce**
1. Create project with structure above
2. Set `pre-commit-config.yml` to something like this:
```
repos:
  - repo: https://github.com/dbt-checkpoint/dbt-checkpoint
    rev: dd74d50dfa32617b1ddaa8f745c3dbd0d05c13e4
    hooks:
      - id: check-model-has-meta-keys
        args:
          [
            "--allow-extra-keys",
            "--meta-keys",
            "a",
            "b",
            "c",
            "d",
            "--",
          ]
```
3. `dbt run --select @project-name.dir1.sub_dir1`
4. `pre-commit run check-model-has-meta-keys --files models/dir1/sub_dir1/model_a.sql`

**Expected behavior**
I expected it to succeed or fail based on the meta keys and not crash.

**Version:**
commit: dd74d50dfa32617b1ddaa8f745c3dbd0d05c13e4

**Additional context**
I believe it's failing on this line specifically of the `check_model_has_meta_keys.py`
```python
    schemas = get_model_schemas(list(ymls.values()), filenames)
```

which in turn calls:
```python
def get_model_schemas(
    yml_files: Sequence[Path], filenames: Set[str], all_schemas: bool = False
) -> Generator[ModelSchema, None, None]:
    for yml_file in yml_files:
        with open(yml_file, "r") as file:
            schema = yaml.safe_load(file)
            for model in schema.get("models", []):
                if isinstance(model, dict) and model.get("name"):
                    model_name = model.get("name", "")  # pragma: no mutate
                    if model_name in filenames or all_schemas:
                        yield ModelSchema(
                            model_name=model_name,
                            file=yml_file,
                            filename=yml_file.stem,
                            schema=model,
                        )
```
Problem is that the value in the `ymls` variable in the `has_meta_key()` function has the following structure:

```
{'models_schema': PosixPath(project-name/models/dir1/sub_dir1/models_schema.yml')}
```

But when I print the `sqls` variable I get this:

```
{'model_a': PosixPath('models/dir1/sub_dir1/model_a.sql')}
```

Why the difference? Why does one have the `project-name` prefixing the path and one doesn't?

Let me know if this helps?

**Comment by sfc-gh-pkommini** [comment]
cc @tomsej @ssassi @BAntonellini Could you please advise? Is there any quick fix that I can do to help?

**Comment by sfc-gh-pkommini** [comment]
#99 should fix the issue. @tomsej @BAntonellini @ssassi Do you see any issues with this?

**Comment by sfc-gh-pkommini** [comment]
Bug was fixed in PR #105.
